### PR TITLE
Simplify the CLI implementation and improve documentation

### PR DESCRIPTION
**Pull request by TheEdgeOfRage**
Close T-18169

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -2,43 +2,56 @@ package main
...
-		if !strings.HasPrefix(err.Error(), "Usage") {
-			os.Exit(1)
-		}
-		return
-	}
-	if (cfg.ExecutionID == "" && cfg.QueryID == 0) || (cfg.ExecutionID != "" && cfg.QueryID != 0) {
+	queryID := flag.Int("q", 0, "The ID of the query to execute. Conflicts with -e")
+	queryParametersStr := flag.String("p", "{}", "Parameters to pass to the query in JSON format")
+	executionID := flag.String("e", "", "ID of an existing execution to check status. Conflicts with -q")
+	maxRetries := flag.Int("max-retries", 5, "Max number of get errors tolerated before giving up")
```
Comment (RichardKeo) [inline cmd/main.go:18]:
```suggestion
	maxRetries := flag.Int("max-retries", 5, "Max number of errors tolerated before giving up")
```

File: cmd/main.go [hunk 2]
```diff
@@ -2,43 +2,56 @@ package main
...
+	pollInterval := flag.Duration("poll-interval", 5*time.Second, "Interval in seconds for polling for results")
+
+	flag.Parse()
+
+	if (*executionID == "") == (*queryID == 0) {
```
Comment (RichardKeo) [inline cmd/main.go:23]: nit: clever but makes it difficult to read for the reader (or at least my small brain)

File: config/config.go [hunk 3]
```diff
@@ -1,38 +1,43 @@
...
-	var env Env
-	parser := flags.NewParser(&env, flags.Default)
-	_, err := parser.Parse()
-	return env, err
+func ParseEnv() (*Env, error) {
+	apiKey, err := getenvOrError("DUNE_API_KEY")
+	if err != nil {
+		return nil, err
+	}
+	host := getenvOrDefault("DUNE_API_HOST", "https://api.dune.com")
```
Comment (RichardKeo) [inline config/config.go:37]: It would probably be worth moving the default host to a const/var at the top of the file

**Review by RichardKeo — APPROVED** [review]

File: cmd/main.go (hunk 2 above)
Comment (TheEdgeOfRage) [reply cmd/main.go:23]: :smile: I previously had this as `(cfg.ExecutionID == "" && cfg.QueryID == 0) || (cfg.ExecutionID != "" && cfg.QueryID != 0)`, and it annoyed me that it's so long. But I guess you're right, readability over simplicity.

**Pull request merged**